spring-integration-redis/src/main/java/org/springframework/integration/redis/util/RedisLockRegistry.java

```diff
@@ -287,7 +333,11 @@ private boolean obtainLock() {
 			boolean result = Boolean.TRUE.equals(success);
 
 			if (result) {
-				this.lockedAt = System.currentTimeMillis();
+				final long now = System.currentTimeMillis();
+				synchronized (RedisLockRegistry.this.locks) {
```

Comment:
No, this is not OK.
Better to have a heavy `expireUnusedOlderThan()` iterating over all the entries, than have a block on every single `Lock.lock()`.

Reply:
If it is not sorted when `RedisLockRegistry.RedisLock.obtainLock()` is called, the Lock object that was recently called `RedisLockRegistry.RedisLock.lock()` in `removeEldestEntry()` may be removed.

Is this ok because the cache has been removed?

Reply:
I think it is not a big deal, since the `Lock` object is still valid.
We only miss it in the cache for the next `obtain()` call.
I think more like this cache feature should go at all, but it won't heart to have the logic like we do right now: it is very rear situation that some project would have more then 1,000,000 locks in its logic.

Reply:
thanks for your comment
Unfortunately, I recently discovered that it is used in a similar case in another project.
It is being used to prevent simultaneous execution of a specific request by a user.

rolled back `obtainLock()`